pkg/proxy/proxier.go

```diff
@@ -791,3 +793,7 @@ func (proxier *Proxier) iptablesHostPortalArgs(destIP net.IP, destPort int, prot
 	args = append(args, "-j", "DNAT", "--to-destination", net.JoinHostPort(proxyIP.String(), strconv.Itoa(proxyPort)))
 	return args
 }
+
+func isClosedError(err error) bool {
+	return strings.Contains(err.Error(), "use of closed network connection")
```

Comment:
Using strings to detect error conditions makes me very uncomfortable.  Is this really required?  If we really want to avoid the error we should use a channel or something to signal that the proxyLoop should exit.  But I am not sure it's really necessary